src/pages/workspaces/workspace/WorkspaceMenu.js

```diff
@@ -0,0 +1,90 @@
+import { Fragment } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { Clickable } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { makeMenuIcon, MenuButton, MenuTrigger } from 'src/components/PopupTrigger'
+import { useWorkspaceDetails } from 'src/components/workspace-utils'
+import * as Utils from 'src/libs/utils'
+
+
+// In `workspaceInfo`, specify either `name and namespace` to fetch the Workspace details,
+// or `canShare, isAzureWorkspace, isLocked, and isOwner` to use previously fetched details.
+const WorkspaceMenu = ({
+  iconSize, popupLocation,
+  callbacks: { onClone, onShare, onLock, onDelete },
+  workspaceInfo: { name, namespace, canShare, isAzureWorkspace, isLocked, isOwner }
+}) => {
+  const navIconProps = {
+    style: { opacity: 0.65, marginRight: '1rem' },
+    hover: { opacity: 1 }, focus: 'hover'
+  }
+
+  const menuContent = !!namespace ?
+    h(DynamicWorkspaceMenuContent, { namespace, name, onShare, onClone, onDelete, onLock }) :
+    h(WorkspaceMenuContent, { canShare, isAzureWorkspace, isLocked, isOwner, onClone, onShare, onLock, onDelete })
+
+  return h(MenuTrigger, {
+    side: popupLocation,
+    closeOnClick: true,
+    content: menuContent
+  }, [
+    h(Clickable, {
+      'aria-label': !!name ? `Action Menu for Workspace: ${name}` : 'Workspace Action Menu',
```

Comment:
I added "Action" to try to clarify what the menu is… but could certainly drop it if folks think it's unnecessary.